Formula/swift.rb

```diff
@@ -40,6 +40,7 @@ class Swift < Formula
   uses_from_macos "llvm" => [:build, :test]
   uses_from_macos "rsync" => :build
   uses_from_macos "curl"
+  uses_from_macos "icu4c" # Used in swift-corelibs-foundation
```

Comment:
I think the macOS build doesn't use `libicu` here.

Reply:
Yes, that's correct. swift-corelibs-foundation is only built on Linux, so the Linux-only `depends_on` is semantically more correct here.

Reply:
I added `swift` to `provided_by_macos_depends_on_allowlist.json` but do you think a new allowlist should be created?